tsdb/index/tsi1/partition.go

```diff
@@ -0,0 +1,1179 @@
+	fs := i.RetainFileSet()
+	defer fs.Release()
+
+	itr := fs.MeasurementIterator()
+	if itr == nil {
+		return nil, nil
+	}
+
+	var a [][]byte
+	for e := itr.Next(); e != nil; e = itr.Next() {
+		if re.Match(e.Name()) {
+			// Clone bytes since they will be used after the fileset is released.
+			a = append(a, bytesutil.Clone(e.Name()))
+		}
+	}
+	return a, nil
+}
+
+func (i *Partition) MeasurementSeriesIDIterator(name []byte) (tsdb.SeriesIDIterator, error) {
+	fs := i.RetainFileSet()
+	return newFileSetSeriesIDIterator(fs, fs.MeasurementSeriesIDIterator(name)), nil
+}
+
+// DropMeasurement deletes a measurement from the index.
+func (i *Partition) DropMeasurement(name []byte) error {
+	fs := i.RetainFileSet()
+	defer fs.Release()
+
+	// Delete all keys and values.
+	if kitr := fs.TagKeyIterator(name); kitr != nil {
+		for k := kitr.Next(); k != nil; k = kitr.Next() {
+			// Delete key if not already deleted.
+			if !k.Deleted() {
+				if err := func() error {
+					i.mu.RLock()
+					defer i.mu.RUnlock()
+					return i.activeLogFile.DeleteTagKey(name, k.Key())
+				}(); err != nil {
+					return err
+				}
+			}
+
+			// Delete each value in key.
+			if vitr := k.TagValueIterator(); vitr != nil {
+				for v := vitr.Next(); v != nil; v = vitr.Next() {
+					if !v.Deleted() {
+						if err := func() error {
+							i.mu.RLock()
+							defer i.mu.RUnlock()
+							return i.activeLogFile.DeleteTagValue(name, k.Key(), v.Value())
+						}(); err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// Delete all series in measurement.
+	if sitr := fs.MeasurementSeriesIDIterator(name); sitr != nil {
+		defer sitr.Close()
+
+		for {
+			s, err := sitr.Next()
+			if err != nil {
+				return err
+			} else if s.SeriesID == 0 {
+				break
+			}
+			if !fs.SeriesFile().IsDeleted(s.SeriesID) {
+				if err := func() error {
+					i.mu.RLock()
+					defer i.mu.RUnlock()
+					return i.sfile.DeleteSeriesID(s.SeriesID)
+				}(); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	// Mark measurement as deleted.
+	if err := func() error {
+		i.mu.RLock()
+		defer i.mu.RUnlock()
+		return i.activeLogFile.DeleteMeasurement(name)
+	}(); err != nil {
+		return err
+	}
+
+	// Check if the log file needs to be swapped.
+	if err := i.CheckLogFile(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createSeriesListIfNotExists creates a list of series if they doesn't exist in
+// bulk.
+func (i *Partition) createSeriesListIfNotExists(names [][]byte, tagsSlice []models.Tags) error {
+	// Maintain reference count on files in file set.
+	fs := i.RetainFileSet()
+	defer fs.Release()
+
+	// Ensure fileset cannot change during insert.
+	i.mu.RLock()
+	// Insert series into log file.
+	if err := i.activeLogFile.AddSeriesList(names, tagsSlice); err != nil {
+		i.mu.RUnlock()
+		return err
+	}
+	i.mu.RUnlock()
+
+	return i.CheckLogFile()
+}
+
+func (i *Partition) DropSeries(key []byte, ts int64) error {
+	// TODO: Use ts.
```

Comment:
It looks like the `ts` change to prevent index inconsistency when writing and dropping the same series concurrently is gone.  We need to handle that or that delete bug will be back.